Plots/Polygons/NCL_polyg_2.py

```diff
@@ -0,0 +1,155 @@
+
+See following URLs to see the reproduced NCL plot & script:
+    - Original NCL script: https://www.ncl.ucar.edu/Applications/Scripts/polyg_2.ncl
+    - Original NCL plot: https://www.ncl.ucar.edu/Applications/Images/polyg_2_lg.png
+"""
+
+###############################################################################
+# Import packages:
+# ----------------
+import numpy as np
+import xarray as xr
+import cartopy.crs as ccrs
+import matplotlib.pyplot as plt
+import matplotlib.colors as colors
+import matplotlib.cm as cm
+from geocat.viz import util as gvutil
+
+import geocat.datafiles as gdf
+import cartopy.io.shapereader as shpreader
+import shapely.geometry as sgeom
+from mpl_toolkits.axes_grid1.inset_locator import inset_axes
+
+###############################################################################
+# Set state data and the color map dictionary
+
+colormap = {1: 'mediumpurple', 2: 'mediumblue', 3: 'royalblue',
+            4: 'cornflowerblue', 5: 'lightblue', 6: 'teal', 7: 'yellowgreen', 8: 'green',
+            9: 'wheat', 10: 'tan', 11: 'gold', 12: 'orange', 13: 'red', 14: 'firebrick'}
+
+###############################################################################
+# Plot map and colorbar
+
+# Create plot figure
+fig = plt.figure(figsize=(10, 10))
+
+# Add a subplot for lambert conformal map
+ax = fig.add_subplot(111, projection=ccrs.LambertConformal(), frameon=False, xbound=0.0, ybound=0.0)
+
+# Set latitude and longitude extent of map
+ax.set_extent([-119, -74, 18, 50], ccrs.Geodetic())
+
+# Set shape name of map (which depicts the United States)
+shapename = 'admin_1_states_provinces_lakes_shp'
+states_shp = shpreader.natural_earth(resolution='110m', category='cultural', name=shapename)
+
+# Set title and title fontsize of plot using gvutil function instead of matplotlib function call
+gvutil.set_titles_and_labels(ax, maintitle="\nAverage Annual Precipiation \n Computed for the period 1899-1999 \n NCDC climate division data \n \n",
+                            maintitlefontsize=18)
+
+# Add outlines of each state within the United States
+for state in shpreader.Reader(states_shp).geometries():
+
+    facecolor = 'white'
+    edgecolor = 'black'
+
+    ax.add_geometries([state], ccrs.PlateCarree(), facecolor=facecolor, edgecolor=edgecolor)
+
+# Open climate division datafile and add to xarray
+ds = xr.open_dataset(gdf.get("netcdf_files/climdiv_prcp_1899-1999.nc"), decode_times=False)
+
+# For each variable (climate division) in data set, create outline on map and fill with random color
+for varname, da in ds.data_vars.items():
+
+    try:
+        # Get precipitation data for each climate division
+        first = ds.get(varname)
+        precipitationdata = sum(first.values)/100
+
+        # Get borders of each climate division
+        lat = first.lat
+        lon = first.lon
+
+        # Determine which color to fill the divisions based on precipitation data
+        colorindex = 0
+        if (precipitationdata<5):
```

Comment:
Rather than having a big, hard-coded if/else statement, how about writing a small helper function that take a 'bounds' array and a value (precipitation data), and returns a color index. It will make the code more maintainable and extendable if the mapping ever needs to be changed. Also you won't have all of the special bounds values duplicated (once in the loop, and once below in the 'bounds' array)

Reply:
Nice!